src/EditorServicesCommandSuite/CodeGeneration/Write.cs

```diff
@@ -8,6 +8,11 @@ internal static class Write
     {
         internal static void AsExpressionValue(PowerShellScriptWriter writer, ParameterBindingResult element)
         {
+            if (element == null)
+            {
+                return;
```

Comment:
If we're good on the other hint direction, I would change `ParameterBindingResult` to the parameter class you made and  replace the logic here to write the new hint style.

Reply:
I'm uncertain what your line of thought is here. So we'll change the hints to dummy variables. And we can add the logic for them here. Clear so far.

But why change signature of this method, and make an internal method use a private type as input? 
Also, the logic to extract the Value/text from the ParameterBindingResult will have to move elsewhere, arguably to the Parameter class. But what use does this class have after that, why not move the method to the PowershellWriter class e.g.?


- Write.AsExpressionValue()